### ISSUE TITLE
Optional DateTime always return TRUE for `some(myDate)`

### ISSUE THREAD

**Issue by martinhalford** [accordproject/ergo#851]
# Bug Report 🐛
Please see LOOM video for demo of bug.
https://www.loom.com/share/4d628fd1e1ec4afa9d5a50444e69ebe9

In summary, `optional` variables return unexpected results when interrogated in code blocks within `grammar.tem.md` files.

Example CTA archive attached:
[OptionalDateTime CTA](https://github.com/accordproject/ergo/files/8140406/OptionalDateTime.zip)

**Comment by ritikBhandari** [comment]
Is this issue still prevalent? Are you still facing it?

**Comment by prekel** [comment]
You shouldn't match on `some(myDate)`, because in this case expression `some(myDate)` will always has value and `completionDate` will have type `DateTime optional`. Your code snipped is equivalent to just 
```
{{%
    "YES - Found myDate - " ++ toString(myDate)
%}}
```

Match should be done on `myDate` itself:
```
{{%
    match myDate
    with let? completionDate then "YES - Found myDate - " ++ toString(completionDate)
    else "NO - myDate not found"
%}}
```

Converting optional value to string is quite ugly, yes.

**Comment by shubhamkumar13** [comment]
I agree with @prekel here some(MyDate) shouldn't be pattern matched here.

@prekel can you please create a PR (if you are willing to) for the tests that you added to the your fork I think that would be helpful in avoiding this :smiley:
